modules/test/playwright/pages/object-web/list-type/ListTypeDefinitionsPage.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+import {ApplicationsMenuPage} from '../../product-navigation-applications-menu/ApplicationsMenuPage';
+
+export class ListTypeDefinitionsPage {
+	readonly addPicklistButton: Locator;
+	readonly applicationsMenuPage: ApplicationsMenuPage;
+	readonly page: Page;
+	readonly picklistNameInput: Locator;
+	readonly savePicklistButton: Locator;
+
+	constructor(page: Page) {
+		this.addPicklistButton = page
+			.getByRole('button', {
+				name: 'Add Picklist',
+			})
+			.first();
+		this.applicationsMenuPage = new ApplicationsMenuPage(page);
+		this.page = page;
+		this.picklistNameInput = page.locator(
+			'input[id="listTypeDefinitionName"]'
+		);
+		this.savePicklistButton = page.getByRole('button', {
+			name: 'Save',
+		});
+	}
+
+	async createPicklist(picklistName: string) {
+		await this.addPicklistButton.click();
+		await this.picklistNameInput.click();
+		await this.picklistNameInput.fill(picklistName);
+
+		const responsePromise = this.page.waitForResponse(
+			'**/list-type-definitions'
+		);
+		await this.savePicklistButton.click();
+
+		const response = await responsePromise;
```

Comment:
I see that this is not working because of the page reload. I thought about two different options here. The first one is to simply wait for the link in FDS to be visible, like this:

```
await this.page.getByRole('link', {name: picklistName}).waitFor({state: 'visible'});
```
Then make the necessary GET calls to the API and filter by name to get the ID. I'd rather do that on a test then add a timeout in the application. Its better to degrade a little bit of the test performance than to degrade a little bit of the application performance.

The second one is to get the Pick list ID from the link in FDS, like this:

```
async createPicklist(picklistName: string) {
	await this.addPicklistButton.click();
	await this.picklistNameInput.click();
	await this.picklistNameInput.fill(picklistName);

	await this.savePicklistButton.click();

	const picklistLink = this.page.getByRole('link', {name: picklistName});

	await picklistLink.waitFor({state: 'visible'});

	const linkHref = new URL(await picklistLink.getAttribute('href'));

	return linkHref.searchParams.get('_com_liferay_object_web_internal_list_type_portlet_portlet_ListTypeDefinitionsPortlet_listTypeDefinitionId');
}
```

`'_com_liferay_object_web_internal_list_type_portlet_portlet_ListTypeDefinitionsPortlet_listTypeDefinitionId'` should be stored as a constant in another file. This will prevent an extra API call by using information already available in the DOM. To be honest, I prefer doing the extra API call. Its is more resilient to changes in the DOM, though these portlet paths are very unlikely to change and we already [rely on them for accessing portlets directly](https://github.com/liferay/liferay-portal/blob/master/modules/test/playwright/utils/portletUrls.ts).

Reply:
ok! i'll try to make the way you suggested using the GET calls

Reply:
I talked to @igor-franca and he gave two more suggestions:

1. Just delete every non-system picklistt
2. do the API call, but filter by name by passing a `?filter=` param in the URL ([docs](https://help.liferay.com/hc/en-us/articles/360031163631-Filter-Sort-and-Search))